lib/phoenix/endpoint/supervisor.ex

```diff
@@ -118,10 +118,12 @@ defmodule Phoenix.Endpoint.Supervisor do
     end
   end
 
-  defp socket_children(endpoint, fun) do
+  defp socket_children(otp_app, endpoint, fun) do
     for {_, socket, opts} <- Enum.uniq_by(endpoint.__sockets__(), &elem(&1, 1)),
         spec = apply_or_ignore(socket, fun, [[endpoint: endpoint] ++ opts]),
```

Comment:
Since this is calling `start_link` in the transport, can't you move the check to the transport's `start_link` instead? This way we only check it for the transports that make sense.

Reply:
Ah, ignore me, the transport is not defined in Phoenix anymore. :)